src/minerva_lib/blend.py

```diff
@@ -1,102 +1,70 @@
 import numpy as np
+import skimage.exposure
 
 
-def to_f32(img):
-    '''Scale the dynamic range to 0.0 - 1.0
-
-    Arguments:
-    img: An integer image
+def handle_channel(channel):
     '''
-
-    # No well-defined behavior for decimal values or values less than 0
-    try:
-        dtype_info = np.iinfo(img.dtype)
-        assert dtype_info.min is 0
-    except (ValueError, AssertionError):
-        raise ValueError('Scaling to 0,1 requires unsigned integers')
-
-    one = dtype_info.max
-    return np.float32(img / one)
-
-
-def f32_to_bgr(f_img, color=[1, 1, 1]):
-    '''Reshape into a color image
-
     Arguments:
-    f_img: float32 image to reshape
+        channel: Dict to blend with rendering settings:
+            {
+                image: Numpy image data
+                min: Range minimum, float32 range 0, 1
+                max: Range maximum, float32 range 0, 1
+            }
+
+    Returns:
+        Threshholded float32 image normalzied within 0, 1
+        r, g, b float32 array color within 0, 1
     '''
+    image = channel['image']
+    min_ = channel['min']
+    max_ = channel['max']
 
-    # All inputs should be normalized between 0 and 1
-    try:
-        assert np.all((f_img >= 0) & (f_img <= 1))
-    except AssertionError:
-        raise ValueError('Color image requires values from 0,1')
+    # Return image for additive blending
+    f32_image = skimage.img_as_float(image)
+    f32_image = skimage.exposure.rescale_intensity(f32_image, (min_, max_))
 
-    # Give the image a color dimension
-    f_vol = f_img[:, :, np.newaxis]
-    f_bgr = np.repeat(f_vol, 3, 2) * color
-    return np.round(255 * f_bgr).astype(np.uint8)
+    return f32_image, channel['color']
```

Comment:
One problem here is only relevant for testing. Due to `skimage.img_as_float`, `f32_image` will not always be a 2D array. If `image=np.uint16([[0]])`, then `f32_image == np.float32(0)`. It does not equal `np.float32([[0]])`. Basically, singleton dimensions are removed when converting to a float32 array. Since we don't have code here to convert all the outputs here to 2D arrays, we later need a more complicated method for [calculating the size of the output buffer](https://github.com/sorgerlab/minerva-lib-python/pull/17/files#diff-b4a15c8818fcee8c893d7b67f6fd3b42R53).

Reply:
It's actually `skimage.exposure.rescale_intensity` that causes the dimensionality reduction problem. And this problem only occurs on `np.float64` arrays, which are returned by `skimage.img_as_float`. We're just deciding not to support images of exactly one pixel by one pixel in shape.